### PR TITLE
Change datetime to datetime-local helper tag

### PR DESCRIPTION
**Pull request by herminiotorres**
A change was made in the helper that renders the `datetime`,
being now by default `datetime-local` and creating
an alias of `datetime-local` for `datetime`, `datetime` tag and
it passes to be an abstract class for all other tags that inherit from him.

As a new specification of the HTML 5 the text field type `datetime`
will no longer exist and will pass a `datetime-local`.
Ref: https://html.spec.whatwg.org/multipage/forms.html#local-date-and-time-state-(type=datetime-local)

### CODE REVIEW

**Comment by maclover7** [comment]
`datetime_field_tag` and `datetime_field` are both being deprecated in Rails 5.1 (the current Rails master branch), so it'd be better to avoid touching those, since they are going away very soon 😬

**Comment by rafaelfranca** [comment]
He is pair programming with me, I asked this change. I prefer to keep `datetime_field` because it is used by simple form as the default datetime method. See https://github.com/plataformatec/simple_form/issues/1394

File: actionview/lib/action_view/helpers/form_helper.rb [hunk 1]
```diff
@@ -1074,68 +1074,37 @@ def time_field(object_name, method, options = {})
...
-      #   datetime_local_field("user", "born_on", min: "2014-05-20T00:00:00")
+      #   datetime_field("user", "born_on", min: "2014-05-20T00:00:00")
       #   # => <input id="user_born_on" name="user[born_on]" type="datetime-local" min="2014-05-20T00:00:00.000" />
       #
-      def datetime_local_field(object_name, method, options = {})
+      def datetime_field(object_name, method, options = {})
         Tags::DatetimeLocalField.new(object_name, method, self, options).render
       end
 
+      alias datetime_local_field datetime_field
```
Comment (maclover7) [inline actionview/lib/action_view/helpers/form_helper.rb]: Do we need to change the deprecation shipping with 5.0 if we are not deprecating this, and `alias`ing instead? 😬

**Comment by shakycode** [comment]
Thanks for this!  Definitely needed.

**Pull request merged**